### PR TITLE
feat(TDI-44915) access token feature

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
There are only HMAC and Service Account file authentication types.
https://jira.talendforge.org/browse/TDI-44915

**What is the new behavior?**
Provide a new way of OAuth2 Token access.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/tBigQueryInput_begin.javajet [hunk 1]
```diff
@@ -195,17 +199,23 @@
...
+<%} else {
+
+  if (ElementParameterParser.canEncrypt(node, "__ACCESS_TOKEN__")) {%>
+     final String decryptedAccessToken_<%=cid%> = routines.system.PasswordEncryptUtil.decryptPassword(<%=ElementParameterParser.getEncryptedValue(node, "__ACCESS_TOKEN__")%>);
+  <%} else {%>
+     final String decryptedAccessToken_<%=cid%> = <%= ElementParameterParser.getValue(node, "__ACCESS_TOKEN__")%>;
+  <%}%>
+		credential_<%=cid%> =
+			new com.google.api.client.auth.oauth2.Credential(com.google.api.client.auth.oauth2.BearerToken.authorizationHeaderAccessMethod()).setFromTokenResponse(
+				(new com.google.api.client.auth.oauth2.TokenResponse()).setAccessToken(decryptedAccessToken_<%=cid%>));
```
Comment (chmyga) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/tBigQueryInput_begin.javajet:216]: Don't we need to update this token from function(if it is set as function) in BigQuery Input?

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQuerySQLRow/tBigQuerySQLRow_begin.javajet [hunk 2]
```diff
@@ -151,17 +155,27 @@
...
+    <% } else {
+
+      if (ElementParameterParser.canEncrypt(node, "__ACCESS_TOKEN__")) {%>
+         final String decryptedAccessToken_<%=cid%> = routines.system.PasswordEncryptUtil.decryptPassword(<%=ElementParameterParser.getEncryptedValue(node, "__ACCESS_TOKEN__")%>);
+      <%} else {%>
+         final String decryptedAccessToken_<%=cid%> = <%= ElementParameterParser.getValue(node, "__ACCESS_TOKEN__")%>;
+      <%}%>
+    		credential_<%=cid%> =
+    			new com.google.api.client.auth.oauth2.Credential(com.google.api.client.auth.oauth2.BearerToken.authorizationHeaderAccessMethod()).setFromTokenResponse(
+    				(new com.google.api.client.auth.oauth2.TokenResponse()).setAccessToken(decryptedAccessToken_<%=cid%>));
```
Comment (chmyga) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQuerySQLRow/tBigQuerySQLRow_begin.javajet:169]: For BigQueryOutput this token should be updated as well, I guess.

Comment (chmyga) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQuerySQLRow/tBigQuerySQLRow_begin.javajet:169]: Refresh the token from function.

File: main/plugins/org.talend.designer.components.localprovider/components/tGSPut/tGSPut_java.xml [hunk 3]
```diff
@@ -158,16 +187,16 @@
...
 
     <ADVANCED_PARAMETERS>
-	
+
 		<PARAMETER
                 NAME="PART_SIZE"
                 FIELD="TEXT"
                 NUM_ROW="1"
-				SHOW_IF="((USE_EXISTING_CONNECTION == 'true') AND (#LINK@NODE.CONNECTION.AUTH_TYPE == 'SERVICE_ACCOUNT')) OR ((USE_EXISTING_CONNECTION == 'false') AND (AUTH_TYPE == 'SERVICE_ACCOUNT'))"
+				SHOW_IF="((AUTH_TYPE == 'SERVICE_ACCOUNT') OR (AUTH_TYPE == 'TOKEN'))"
```
Comment (chmyga) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSPut/tGSPut_java.xml:195]: Why did you remove check from existing connection? it should be shown if existing connection also uses service account authentication(or oauth token as well, I guess).

**Review by chmyga — CHANGES_REQUESTED** [review]
Some changes are required. Also, we need to find a way to use function for GS components.

File: main/plugins/org.talend.designer.components.localprovider/components/tGSPut/tGSPut_java.xml (hunk 3 above)
Comment (mbasiuk-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSPut/tGSPut_java.xml:195]: agree. I missed this. Gonna update this.

**Review by dmytrogrygorenko — APPROVED** [review]
LGTM

**Review by dmytrogrygorenko — APPROVED** [review]
Still OK.

**Review by dmytrogrygorenko — APPROVED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tGSDelete/tGSDelete_end.javajet [hunk 4]
```diff
@@ -19,16 +19,19 @@ imports="
...
-		}
-	}
-
+	boolean useHMACAuthType = false;
+	if(("true").equals(useExistingConn)) {
+		List< ? extends INode> nodes =  node.getProcess().getNodesOfType("tGSConnection");
+		for (INode ne : nodes) {
+			if (connection.equals(ne.getUniqueName())) {
+				String authTypeInConnectionComponent = ElementParameterParser.getValue(ne, "__AUTH_TYPE__");
+				useHMACAuthType = "USER_ACCOUNT_HMAC".equals(authTypeInConnectionComponent);
```
Comment (dmytrogrygorenko) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSDelete/tGSDelete_end.javajet:28]: Maybe add "break" after this, so no need to iterate further?

File: main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_end.javajet [hunk 5]
```diff
@@ -21,16 +21,19 @@ imports="
...
-		}
-	}
-
+        boolean useHMACAuthType = false;
+        if(("true").equals(useExistingConn)) {
+            List< ? extends INode> nodes =  node.getProcess().getNodesOfType("tGSConnection");
+            for (INode ne : nodes) {
+                if (connection.equals(ne.getUniqueName())) {
+                        String authTypeInConnectionComponent = ElementParameterParser.getValue(ne, "__AUTH_TYPE__");
+                        useHMACAuthType = "USER_ACCOUNT_HMAC".equals(authTypeInConnectionComponent);
```
Comment (dmytrogrygorenko) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_end.javajet:30]: Same here.

File: main/plugins/org.talend.designer.components.localprovider/components/tGSList/tGSList_end.javajet [hunk 6]
```diff
@@ -19,16 +19,19 @@ imports="
...
-    }
-  }
-
+    boolean useHMACAuthType = false;
+    if(("true").equals(useExistingConn)) {
+        List< ? extends INode> nodes =  node.getProcess().getNodesOfType("tGSConnection");
+        for (INode ne : nodes) {
+                if (connection.equals(ne.getUniqueName())) {
+                        String authTypeInConnectionComponent = ElementParameterParser.getValue(ne, "__AUTH_TYPE__");
+                        useHMACAuthType = "USER_ACCOUNT_HMAC".equals(authTypeInConnectionComponent);
```
Comment (dmytrogrygorenko) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSList/tGSList_end.javajet:28]: And here.

**Review by dmytrogrygorenko — CHANGES_REQUESTED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tGSDelete/tGSDelete_end.javajet (hunk 4 above)
Comment (mbasiuk-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSDelete/tGSDelete_end.javajet:28]: The main idea here was to change conditional property. As you see from the original code that was working before, it doesn't introduce a new issue. As for performance - it won't be even seen. (how many tGSConnection components can be present in the single job?)

**Review by dmytrogrygorenko — APPROVED** [review]
LGTM

**Pull request merged**